apisix/core/config_util.lua

```diff
@@ -114,7 +114,7 @@ function _M.fire_all_clean_handlers(item)
         clean_handler.f(item)
     end
 
-    item.clean_handlers = nil
+    item.clean_handlers = {}
```

Comment:
if the `clean_handler` is put into the parents (routers) and set it to nil, it will cause error like this:
```
2023/10/10 20:53:16 [error] 1057074#1057074: *2020 lua entry thread aborted: runtime error: /home/liuwei/api7/apisix/apisix/core/config_util.lua:61: attempt to index field 'clean_handlers' (a nil value)
stack traceback:
coroutine 0:
	/home/liuwei/api7/apisix/apisix/core/config_util.lua: in function 'add_clean_handler'
	/home/liuwei/api7/apisix/apisix/upstream.lua:152: in function 'fetch_healthchecker'
	/home/liuwei/api7/apisix/apisix/upstream.lua:342: in function 'set_upstream'
	/home/liuwei/api7/apisix/apisix/init.lua:549: in function 'handle_upstream'
	/home/liuwei/api7/apisix/apisix/init.lua:730: in function 'http_access_phase'
```